js/common.js

```diff
@@ -1372,3 +1372,25 @@ function flashIconButton(button, button_classes, icon_classes, duration) {
       ico.addClass(original_ico_classes);
    }, duration);
 }
+
+/**
+ * 'Debounces' the click event of a button.
+ * For submit buttons, this will prevent the form from being submitted multiple times.
+ * For other buttons, this will disable them for a specified period of time based on the data-debounce-timeout attribute (defaults to 1 second)
+ * @param btn
+ */
+function debounceButton(btn) {
```

Comment:
1. IMHO, same behaviour should be applied to both submit and base buttons.
2. `debounce-timeout` may lead to unexpected results. For the developper, I guess it would be preferable to have a `undebounceButton` function (or any other better name), that could be used to restore the default button state. Indeed if the `data-debounce` is manually set by developper, I guess he will be able to call this function on the most appropriate moment (on ajax success, ajax failure, on error display, ...).
3. Maybe adding a `<span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>` could enhance the UX.

Reply:
If the developer is going to handle manually re-enabling the button, it doesn't make much since to use this function since all it does for non-submit buttons is disable them. They should just add a click listener that disables the button and then re-enable it when needed.

If the timeout isn't desired, I guess this function should only be for submit buttons where we expect the page to reload or change so it never needs re-enabled.

Reply:
It may make sence if this function does more than just disabling the button, for instance displaying a loader or any other visual hint.